File: app/src/main/java/de/jensklingenberg/jkemptysubmitsearchviewexample/MainActivity.java
package de.jensklingenberg.jkemptysubmitsearchviewexample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.SearchView;
import android.util.Log;

public class MainActivity extends AppCompatActivity {
  EmptySubmitSearchView searchView;
  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    searchView= findViewById(R.id.search);
    searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
      @Override public boolean onQueryTextSubmit(String query) {
        Log.d("TAG", "onQueryTextSubmit: "+query);
        return false;
      }

      @Override public boolean onQueryTextChange(String newText) {
        return false;
      }
    });
  }


}

File: app/src/main/java/de/jensklingenberg/jkemptysubmitsearchviewexample/MainActivity.kt
package de.jensklingenberg.jkemptysubmitsearchviewexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.widget.SearchView
import android.util.Log

class MainActivity : AppCompatActivity() {
    lateinit var searchView: EmptySubmitSearchView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        searchView = findViewById(R.id.search)
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                Log.d("TAG", "onQueryTextSubmit: $query")
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                return false
            }
        })
    }


}
